addons/editor/functions/fnc_deepPaste.sqf

```diff
@@ -0,0 +1,221 @@
+    } forEach _itemTypes;
+
+    {
+        _object addWeaponWithAttachmentsCargoGlobal [_x, 1];
+    } forEach _weapons;
+
+    _magazines params ["_magazineTypes", "_magazineCounts"];
+
+    {
+        _object addMagazineCargoGlobal [_x, _magazineCounts select _forEachIndex];
+    } forEach _magazineTypes;
+
+    _backpacks params ["_backpackTypes", "_backpackCounts"];
+
+    {
+        _object addBackpackCargoGlobal [_x, _backpackCounts select _forEachIndex];
+    } forEach _backpackTypes;
+
+    // Handle containers inside the object's inventory
+    private _allContainers = everyContainer _object;
+
+    {
+        _x params ["_type", "_inventory"];
+
+        private _index = _allContainers findIf {_x select 0 == _type};
+        private _container = _allContainers deleteAt _index select 1;
+
+        [_container, _inventory] call _fnc_restoreInventory;
+    } forEach _containers;
+};
+
+private _fnc_createUnit = {
+    params ["_type", "_position", "_direction", "_rank", "_skill", "_stance", "_loadout", "_group", "_isLeader"];
+
+    _position = _position vectorAdd _centerPos;
+    _group = _group call _fnc_createGroup;
+
+    private _unit = _group createUnit [_type, _position, [], 0, "NONE"];
+    _unit setDir _direction;
+    _unit setRank _rank;
+    _unit setSkill _skill;
+    _unit setUnitPos _stance;
+    _unit setUnitLoadout _loadout;
+
+    if (_isLeader) then {
+        _group selectLeader _unit;
+    };
+
+    _createdObjects pushBack _unit;
+
+    _unit
+};
+
+private _fnc_createVehicle = {
+    params ["_type", "_position", "_dirAndUp", "_inventory", "_customization", "_turretMagazines", "_pylonMagazines", "_vehicleCargo", "_crew"];
+
+    _position = _position vectorAdd _centerPos;
+
+    private _placement = ["NONE", "FLY"] select (_type isKindOf "Air" && {_position select 2 > 5});
+    private _vehicle = createVehicle [_type, _position, [], 0, _placement];
+    _vehicle setVectorDirAndUp _dirAndUp;
+
+    [_vehicle, _inventory] call _fnc_restoreInventory;
+
+    _customization params ["_textures", "_animations"];
+    [_vehicle, _textures, _animations, true] call BIS_fnc_initVehicle;
+
+    {
+        _x params ["_magazine", "_turretPath"];
+
+        _vehicle removeMagazineTurret [_magazine, _turretPath];
+    } forEach magazinesAllTurrets _vehicle;
+
+    {
+        _vehicle addMagazineTurret _x;
+    } forEach _turretMagazines;
+
+    // todo: pylons
+
+    {
+        _vehicle setVehicleCargo (_x call _fnc_createVehicle);
+    } forEach _vehicleCargo;
+
+    {
+        _x params ["_unitData", "_role", "_cargoIndex", "_turretPath"];
+
+        private _unit = _unitData call _fnc_createUnit;
+
+        switch (_role) do {
+            case "driver": {
+                if (getText (configFile >> "CfgVehicles" >> typeOf _unit >> "simulation") == "UAVPilot") then {
+                    _unit moveInAny _vehicle; // moveInDriver does not appear to work for virtual UAV crew, moveInAny does
```

Comment:
Will it guarantee that it will always be the first unit? Because the rover UAV has slots for the passengers as well and if the unit is moveInAny’d then it may place the UAV unit in the passenger seat, no?

Reply:
`moveInAny` prioritizes the driver role and the other crew are placed using role specific commands.